core/trino-main/src/main/java/io/trino/execution/QueryStateMachine.java

```diff
@@ -1048,7 +1048,7 @@ private static boolean isScheduled(Optional<StageInfo> rootStage)
         }
         return getAllStages(rootStage).stream()
                 .map(StageInfo::getState)
-                .allMatch(state -> state == StageState.RUNNING || state == StageState.FLUSHING || state.isDone());
+                .allMatch(state -> state == StageState.RUNNING || state == StageState.PENDING || state.isDone());
```

Comment:
update naming in commit message for `Implement full query retries` commit.

Reply:
Naming `SqlStageExecution` vs `PipelinedStageExecution` is not natural to me. 
It feels like those are coming from the same domain, just are different "types" of "stage execution".
What about renaming "SqlStageExecution". StageExecutionTracker maybe?

Can be done in a followup.

Reply:
Yeah, the naming is difficult. What do you think about simply  calling it `SqlStage`?

> Can be done in a followup.

We can do that, or maybe as a separate commit at the end of this PR (to avoid dealing with rebases?)

Reply:
SqlStage seems fine - though I am not sure if it does not clash with sth else.

> to avoid dealing with rebases

Definitelly, rebasing multiple commits on top would be too painful

Reply:
Renamed to `SqlStage`